api/mesh/v1alpha1/proxy_template.proto

```diff
@@ -24,6 +24,107 @@ message ProxyTemplate {
     // List of raw xDS resources.
     // +optional
     repeated ProxyTemplateRawResource resources = 2;
+
+    // List of config modifications
+    repeated Modifications modifications = 3;
+  }
+
+  // Modifications to xDS config generated by Proxy Template
+  message Modifications {
+    oneof type {
+      // Cluster modification
+      Cluster cluster = 1;
+      // Listener modification
+      Listener listener = 2;
+      // Network Filter modification
+      NetworkFilter networkFilter = 3;
+      // HTTP Filter modification
+      HttpFilter httpFilter = 4;
```

Comment:
Is this all? I look here https://www.envoyproxy.io/docs/envoy/latest/api-v2/api, and there are more entities.

Reply:
There are only a couple of fundamental Envoy resources:
* Cluster - it's here
* Listener - it's here
* ClusterLoadAssignment - there might be a use case of adding extra endpoint, but I don't want to implement it for now. I think I'll open an issue for this and maybe someone from the community will pick this up.
* RouteConfiguration - I want to add it in a separate PR
* Secret - I don't think there will be a use case for this
* Runtime - again, no "popular" use case for now

You can see a lot of entities that are embedded in fundamental types. We added modifications for filters because it needs a special treatment (add in proper order). Having a "framework" for modifications we can later easily add other extension points if needed.

Reply:
OK that makes sense.